### PR TITLE
Full names for suggestions.

### LINKED ISSUE #13000: Themes: Display full taxonomy name in showcase, instead of slug
Could we display the full showcase taxonomy name in the showcase, instead of the slug, like earlier showcases did.

Example:

wordpress themes wordpress com-1

The full name is already in WP Admin here:

theme features theme showcase wordpress

The full name is easier for the human eye and brain to scan and read than the slug, which usually includes one or more hyphens.

It also looks more like plain English to the eye, instead of like a code, making it more user-friendly to look at as well.

### PR DESCRIPTION
**Pull request by budzanowski**
### Info 
This PR implements request from https://github.com/Automattic/wp-calypso/issues/13000

image

### Testing 
Go to `/themes`. Suggestions should have full names displayed. Suggesting and suggestions interactions: mous + keyboard should work as previously.

This closes #13000

### CODE REVIEW

**Comment by folletto** [comment]
Can you add a screenshot? ;)

**Comment by folletto** [comment]
Tested in Chrome, works well. :)

screen shot 2017-05-11 at 12 26 00

I'll do some sizing tweaks after this PR lands. :)

**Comment by seear** [comment]
Order has changed:

**Master** 
screen shot 2017-05-11 at 12 44 13

**Branch**
screen shot 2017-05-11 at 12 44 04

**Comment by ockham** [comment]
Tho subject (vertical) first isn't so bad...

**Comment by folletto** [comment]
If I could choose an order, I'd say:

1. Subject
2. Feature
3. Style
4. Column
5. Layout

File: client/components/suggestions/index.jsx [hunk 1]
```diff
@@ -158,7 +160,8 @@ class Suggestions extends React.Component {
 	 * @return {Object}          filtered taxonomy:[ terms ] object
 	 */
 	narrowDownAndSort = ( input, showAll = '' ) => {
-		const [ taxonomy, filter ] = input.split( ':' );
+		const termsTable = mapValues( this.props.terms, term => Object.keys( term ) );
```
Comment (seear) [inline client/components/suggestions/index.jsx]: Can this be
`const termsTable = mapValues( this.props.terms, Object.keys );`
?

Comment (budzanowski) [inline client/components/suggestions/index.jsx]: Ha :) Nice.

**Comment by seear** [comment]
Just a comment: using lodash `keys` instead of `Object.keys` might make the code slightly neater. Not sure if there is any other reason to prefer it.

File: client/components/suggestions/index.jsx (hunk 1 above)
Comment (ockham) [inline client/components/suggestions/index.jsx]: Hmm, isn't there a lodash shorthand for that too? `mapKeys`?

**Comment by kathrynwp** [comment]
Regarding the order, by far the most common way users request themes is by feature (anecdotally of course, I don't have stats). @dcoleonline agrees. :) 

Based on my experiences with how users ask for themes, I'd suggest:

1. Feature
2. Layout 
3. Column
4. Subject
5. Style

But I'm not too fussy as long as Feature is first. ;-)

**Comment by dcoleonline** [comment]
I like @kathrynwp's order above. In my experience, when users start looking for a theme based on subject/style, it doesn't always do what they want.  In that case, we have to go back to features and layout to help them find what they need.

**Comment by folletto** [comment]
The list I made above is from Analytics data tho, so it's current use (last month) in order of priority. It's not weighted priority, just "what comes first" in taxonomy use, so it might change if we weight it.

**Comment by budzanowski** [comment]
image
Default order added.

**Pull request merged**